apis/v1alpha2/validation/httproute_test.go

```diff
@@ -189,134 +175,219 @@ func TestValidateHTTPRoute(t *testing.T) {
 									},
 								},
-								{
-									Type: gatewayv1a2.HTTPRouteFilterRequestMirror,
-									RequestMirror: &gatewayv1a2.HTTPRequestMirrorFilter{
-										BackendRef: gatewayv1a2.BackendObjectReference{
-											Name: specialService,
-											Port: portNumberPtr(8080),
-										},
-									},
-								},
 							},
 						},
 					},
 				},
 			},
-			errCount: 2,
+			errCount: 0,
 		},
 		{
-			name: "valid httpRoute with duplicate ExtensionRef filters",
+			name: "invalid httpRoute Rules backendref filters",
 			hRoute: gatewayv1a2.HTTPRoute{
 				Spec: gatewayv1a2.HTTPRouteSpec{
 					Rules: []gatewayv1a2.HTTPRouteRule{
 						{
-							Matches: []gatewayv1a2.HTTPRouteMatch{
-								{
-									Path: &gatewayv1a2.HTTPPathMatch{
-										Type:  pathMatchTypePtr("Prefix"),
-										Value: utilpointer.String("/"),
-									},
-								},
-							},
-							Filters: []gatewayv1a2.HTTPRouteFilter{
+							BackendRefs: []gatewayv1a2.HTTPBackendRef{
 								{
-									Type: gatewayv1a2.HTTPRouteFilterRequestHeaderModifier,
-									RequestHeaderModifier: &gatewayv1a2.HTTPRequestHeaderFilter{
-										Set: []gatewayv1a2.HTTPHeader{
-											{
-												Name:  "special-header",
-												Value: "foo",
+									Filters: []gatewayv1a2.HTTPRouteFilter{
+										{
+											Type: gatewayv1a2.HTTPRouteFilterRequestMirror,
+											RequestMirror: &gatewayv1a2.HTTPRequestMirrorFilter{
+												BackendRef: gatewayv1a2.BackendObjectReference{
+													Name: testService,
+													Port: pkgutils.PortNumberPtr(8080),
+												},
 											},
 										},
-									},
-								},
-								{
-									Type: gatewayv1a2.HTTPRouteFilterRequestMirror,
-									RequestMirror: &gatewayv1a2.HTTPRequestMirrorFilter{
-										BackendRef: gatewayv1a2.BackendObjectReference{
-											Name: testService,
-											Port: portNumberPtr(8080),
+										{
+											Type: gatewayv1a2.HTTPRouteFilterRequestMirror,
+											RequestMirror: &gatewayv1a2.HTTPRequestMirrorFilter{
+												BackendRef: gatewayv1a2.BackendObjectReference{
+													Name: specialService,
+													Port: pkgutils.PortNumberPtr(8080),
+												},
+											},
 										},
 									},
 								},
-								{
-									Type: "ExtensionRef",
-								},
-								{
-									Type: "ExtensionRef",
-								},
-								{
-									Type: "ExtensionRef",
-								},
 							},
 						},
 					},
 				},
 			},
-			errCount: 0,
+			errCount: 1,
 		},
 	}
+
 	for _, tt := range tests {
-		// copy variable to avoid scope problems with ranges
 		tt := tt
 		t.Run(tt.name, func(t *testing.T) {
-			errs := ValidateHTTPRoute(&tt.hRoute)
-			if len(errs) != tt.errCount {
-				t.Errorf("ValidateHTTPRoute() got %v errors, want %v errors", len(errs), tt.errCount)
+			for index, rule := range tt.hRoute.Spec.Rules {
+				errs := validateHTTPBackendUniqueFilters(rule.BackendRefs, field.NewPath("spec").Child("rules"), index)
```

Comment:
Would this work?
```suggestion
				errs := validateHTTPBackendUniqueFilters(rule.BackendRefs, field.NewPath("spec").Child("rules").Index(index))
```

Reply:
literally, integer "index" exactly the index within rules, easier in our case. leave as it is.